chia/cmds/passphrase_funcs.py

```diff
@@ -42,33 +43,56 @@ def verify_passphrase_meets_requirements(
         raise Exception("Unexpected passphrase verification case")
 
 
-def tidy_passphrase(passphrase: str) -> str:
-    """
-    Perform any string processing we want to apply to the entered passphrase.
-    Currently we strip leading/trailing whitespace.
-    """
-    return passphrase.strip()
+def prompt_to_save_passphrase() -> bool:
+    save: bool = False
+
+    try:
+        if supports_os_passphrase_storage():
+            location: Optional[str] = None
 
+            if sys.platform == "darwin":
+                location = "macOS Keychain"
 
-def prompt_for_new_passphrase() -> str:
+            if location is None:
+                raise ValueError("OS-specific credential store not specified")
+
+            print(
+                "\n"
+                "Your passphrase can be stored in your system's secure credential store. "
+                "Chia services will be able to access your keys without prompting for your passphrase."
+            )
+            save = prompt_yes_no(f"Would you like to save your passphrase to the {location} (y/n) ")
+
+    except Exception as e:
+        print(f"Caught exception: {e}")
+        return False
+
+    return save
+
+
+def prompt_for_new_passphrase() -> Tuple[str, bool]:
     min_length: int = Keychain.minimum_passphrase_length()
     if min_length > 0:
         n = min_length
         print(f"\nPassphrases must be {n} or more characters in length")  # lgtm [py/clear-text-logging-sensitive-data]
     while True:
-        passphrase = tidy_passphrase(getpass("New Passphrase: "))
-        confirmation = tidy_passphrase(getpass("Confirm Passphrase: "))
+        passphrase: str = getpass("New Passphrase: ")
+        confirmation: str = getpass("Confirm Passphrase: ")
+        save_passphrase: bool = False
 
         valid_passphrase, error_msg = verify_passphrase_meets_requirements(passphrase, confirmation)
 
         if valid_passphrase:
-            return passphrase
+            if supports_os_passphrase_storage():
+                save_passphrase = prompt_to_save_passphrase()
+
+            return passphrase, save_passphrase
         elif error_msg:
             print(f"{error_msg}\n")  # lgtm [py/clear-text-logging-sensitive-data]
```

Comment:
there's no return statement here. Was this meant to throw an exception?

Reply:
No, this is an interactive step in the CLI wherein the user is prompted to enter a new passphrase. The while loop continues until a valid (requirements are met) passphrase is entered.